### PR TITLE
[15.0][MIG] website_membership_random_order: Migration to version 15.0

### PR DESCRIPTION
**Pull request by stefan-tecnativa**
cc @Tecnativa TT42051

@pedrobaeza please review!

### CODE REVIEW

File: website_membership_random_order/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            "<int:country_id>",
+            "/members/association/<membership_id>/country/<int:country_id>",
+            "/members/association/<membership_id>/country/<country_name>-"
+            "<int:country_id>/page/<int:page>",
+            "/members/association/<membership_id>/country/<int:country_id>/"
+            "page/<int:page>",
+        ],
+        type="http",
+        auth="public",
+        website=True,
```
Comment (chienandalu) [inline website_membership_random_order/controllers/main.py]: This is missing the `sitemap` argument (https://github.com/odoo/odoo/blob/15.0/addons/website_membership/controllers/main.py#L32)

Anyway @pedrobaeza is there a reason to not do a simpler overriding with

```python
@route()
def members(...)
```

**Review by chienandalu — CHANGES_REQUESTED** [review]
It isn't working for me. The members always appear in the same order. Does it work for you?

**Comment by stefan-tecnativa** [comment]
> It isn't working for me. The members always appear in the same order. Does it work for you?

I've observed that in both 13.0 and 14.0 runbots, the behavios that you are describing happens also for the members that has their state on "free member". But for those who has their state on "paid" this is not happening and they are displayed randomly. 
Searching I've found this, but I'm not sure if it has any relation with our problem: https://github.com/odoo/odoo/blob/15.0/addons/website_membership/controllers/main.py#L44

**Comment by chienandalu** [comment]
True, they're randomized inside their membership category and free members don't get randomized at all. We can live with that IMO

File: website_membership_random_order/models/membership_line.py [hunk 2]
```diff
@@ -16,6 +16,7 @@ def search(self, args, offset=0, limit=None, order=None, count=False):
         won't be able to show all the records due to the pager and each time returning
         random results.
         """
+
```
Comment (pedrobaeza) [inline website_membership_random_order/models/membership_line.py]: Why adding this empty line?

Comment (stefan-tecnativa) [reply website_membership_random_order/models/membership_line.py]: I was debugging and I forgot to remove the line when I removed the debugger method. Changed.

**Review by chienandalu — APPROVED** [review]
:+1:

File: website_membership_random_order/controllers/main.py (hunk 1 above)
Comment (pedrobaeza) [reply website_membership_random_order/controllers/main.py]: I didn't know that such option is available, so if everything is kept, then better to do it that way.

Comment (chienandalu) [reply website_membership_random_order/controllers/main.py]: Here's the reference: 
- https://www.odoo.com/documentation/16.0/developer/reference/backend/http.html?highlight=controller#controllers
- https://www.odoo.com/documentation/16.0/developer/reference/backend/http.html?highlight=route#odoo.http.route

Comment (pedrobaeza) [reply website_membership_random_order/controllers/main.py]: Great, then please do, @stefan-tecnativa

Comment (stefan-tecnativa) [reply website_membership_random_order/controllers/main.py]: Changes done. Thanks for the explanations.

**Comment by pedrobaeza** [comment]
With the latest change, `from odoo import http` should go away.

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**